plugins/modules/ec2_instance.py

```diff
@@ -1332,20 +1333,48 @@ def value_wrapper(v):
     param_mappings = [
         ParamMapper('ebs_optimized', 'EbsOptimized', 'ebsOptimized', value_wrapper),
         ParamMapper('termination_protection', 'DisableApiTermination', 'disableApiTermination', value_wrapper),
+        ParamMapper('security_groups', 'Groups', 'groupSet', value_wrapper),
+        ParamMapper('security_group', 'Groups', 'groupSet', value_wrapper),
         # user data is an immutable property
         # ParamMapper('user_data', 'UserData', 'userData', value_wrapper),
     ]
 
     for mapping in param_mappings:
         if params.get(mapping.param_key) is not None and mapping.instance_key not in skip:
             value = AWSRetry.jittered_backoff()(ec2.describe_instance_attribute)(Attribute=mapping.attribute_name, InstanceId=id_)
-            if params.get(mapping.param_key) is not None and value[mapping.instance_key]['Value'] != params.get(mapping.param_key):
-                arguments = dict(
-                    InstanceId=instance['InstanceId'],
-                    # Attribute=mapping.attribute_name,
+            if mapping.param_key not in ['security_groups', 'security_group']:
+                if params.get(mapping.param_key) is not None and value[mapping.instance_key]['Value'] != params.get(mapping.param_key):
+                    arguments = dict(
+                        InstanceId=instance['InstanceId'],
+                        # Attribute=mapping.attribute_name,
+                    )
+                    arguments[mapping.instance_key] = mapping.add_value(params.get(mapping.param_key))
+                    changes_to_apply.append(arguments)
+            elif bool(params.get(mapping.param_key)) is not False:
```

Comment:
I'm not sure this is the right logic.  None is already caught by the if statement we're nested under.  Are you trying to catch an empty list/string?

IMO an empty list would mean 'remove all SGs' and an empty string should be triggering an error (with a nice clear message).

Reply:
Should we support removing all security groups from an instance?

Reply:
You could (an empty list would mean the default security group should be the only group associated).  I'm also okay with the way you did it though, since it seems less potentially surprising.